crates/ruff_server/src/session/settings.rs

```diff
@@ -21,6 +22,22 @@ pub(crate) struct ResolvedClientSettings {
     #[allow(dead_code)]
     disable_rule_comment_enable: bool,
     fix_violation_enable: bool,
+    // TODO(jane): Remove once editor settings resolution is implemented
+    #[allow(dead_code)]
+    editor_settings: ResolvedEditorSettings,
+}
+
+#[derive(Debug, Default)]
+#[cfg_attr(test, derive(PartialEq, Eq))]
+#[allow(dead_code)] // TODO(jane): Remove once editor settings resolution is implemented
+pub(crate) struct ResolvedEditorSettings {
+    lint_preview: Option<bool>,
+    format_preview: Option<bool>,
```

Comment:
Why use `Option<bool>`? Can we define a default value instead? It would probably be `false` for both `lint_preview` and `format_preview`.

Same question for other settings where it makes sense. I guess I'm missing some context here.

Reply:
The reason why these are optional is because we'll eventually merge them with the project configuration. If we set them to a default value here, we wouldn't have a way of knowing whether that was an actually defined value or just the default value resulting from it being un-set on the client side. I'll clarify this in a comment.